### DISCUSSION TITLE
Alerting on a single document

### DISCUSSION THREAD

**Discussion by rdomloge** [jertel/elastalert2#689] [Q&A] [answered]
I have an ES store with various indices, but I am particularly interested in alerting on 2 indices that store static documents that are periodically updated. They are 'status' documents - an automated process interrogates various systems and updates the status of their synching. Periodically, they get out of sync and the status document would indicate this. 

I would like to create an alert that examines such documents and alerts if they fall behind. They are not timeseries indices; at the moment there are 2 documents - total - which store 2 documents containing status information.

Does ElastAlert work well with documents that do not contain timeseries data?

**Comment by jertel** [comment, marked as answer]
It's not a use case that ElastAlert 2 was designed for, but it might be possible, but probably only if those static document records have a time field. If they do then you might have two options:

1. If the time field is static, meaning it was set at the time the static documents were created, then you would have to use the [ElastAlert Rule Test tool](https://elastalert2.readthedocs.io/en/latest/ruletypes.html#testing-your-rule), in combination with the `scan_entire_timeframe` option and a cron scheduler. The cron scheduler would run the rule test rool once, scan the entire time frame (you'd want to set `timeframe` to something large enough so that 5 years from now it's still able to find that document). Once the rule tester finishes, it will exit and the cron scheduler will wait for the next execution time.

2. If the time field is dynamic, meaning it gets updated every time the static documents are updated, then you *might* be able to use ElastAlert 2 normally. I've never attempted this, so I can't tell you with any confidence that it will work, but it might, since it will be querying Elasticsearch for any records where the `timestamp_field` value falls within the sliding time window that gets updated on each query check cycle.
